plugins/database/src/db_plugin.cpp

```diff
@@ -7225,45 +7223,38 @@ irods::error db_update_pam_password_op(
                 return ERROR( status, "commit failure" );
             }
         } // if !irods_pam_auth_no_extend
-        icatDescramble( passwordInIcat );
-        strncpy( *_irods_password, passwordInIcat, irods_pam_password_len );
+
+        // random_password is the randomly generated password (see while loop below) in a scrambled form.
+        icatDescramble(random_password.data());
+
+        // The descrambled password at time of generation is 50 characters or less (see below).
+        std::strncpy(*_password_buffer, random_password.data(), _password_buffer_size);
+
         return SUCCESS();
     }
 
-    // =-=-=-=-=-=-=-
-    // if the resultant scrambled password has a ' in the
-    // string, this can cause issues on some systems, notably
-    // Suse 12.  if this is the case we will just get another
-    // random password.
-    bool pw_good = false;
-    while ( !pw_good ) {
-        j = 0;
-        get64RandomBytes( rBuf );
-        for ( i = 0; i < 50 && j < irods_pam_password_len - 1; i++ ) {
-            char c;
-            c = rBuf[i] & 0x7f;
-            if ( c < '0' ) {
-                c += '0';
-            }
-            if ( ( c > 'a' && c < 'z' ) || ( c > 'A' && c < 'Z' ) ||
-                    ( c > '0' && c < '9' ) ) {
-                randomPw[j++] = c;
-            }
-        }
-        randomPw[j] = '\0';
+    std::array<char, MAX_PASSWORD_LEN> scrambled_random_password{};
```

Comment:
Need another + 1 here, I believe. Sorry

Reply:
The +1s definitely make me nervous.

Good catch though.

Reply:
Done